### PR TITLE
[Core] Change finished to fightend and move to a normalizer

### PR DESCRIPTION
**Pull request by Dambroda**
#2779

### CODE REVIEW

**Comment by Dambroda** [comment]
_isHandlingEvent was added because finished was fabricated outside of a triggering event. Because that is no longer the case (meaning all fabricated events *are* triggered from another event, even if they dont set event.trigger), the logic is no longer needed to prevent the crash 

And yes I wasn't sure about it either (in terms of putting it into Events). I kinda took a middle ground, but I think that because it's a "core" event and not one from another module it kinda makes sense for it to be in events.

File: src/parser/core/CombatLogParser.test.js [hunk 1]
```diff
@@ -240,13 +239,4 @@ describe('Core/CombatLogParser', () => {
       expect(Object.keys(parser._modules).length).toBeGreaterThan(0);
     });
   });
-  describe('emits static events', () => {
```
Comment (MartijnHols) [inline src/parser/core/CombatLogParser.test.js]: Please maintain this test, it's important that the fightend event being fired is guarded

File: src/parser/shared/normalizers/FightEnd.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
+import EventsNormalizer from 'parser/core/EventsNormalizer';
+
+export const END_EVENT_TYPE = 'fightend';
+
+class FightEnd extends EventsNormalizer {
```
Comment (MartijnHols) [inline src/parser/shared/normalizers/FightEnd.js]: Please add a comment above the class shortly explaining what it does and why, and optionally possible extensions.

File: src/parser/shared/normalizers/FightEnd.js [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
```
[the remark this answers is not in the document]
Comment (Dambroda) [reply src/parser/shared/normalizers/FightEnd.js]: no

**Pull request merged**